enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -0,0 +1,878 @@
+
+## Motivation
+
+cert-manager enables obtaining certificates from [various](https://docs.openshift.com/container-platform/4.15/security/cert_manager_operator/index.html#cert-manager-issuer-types_cert-manager-operator-about)
+certificate authorities and also ensures the validity of certificates by renewing them before they expire.
+`OpenShift Service Mesh` secures intra-cluster and external service connections with TLS encryption. `cert-manager`
+can be used to obtain the required certificates through custom resources `issuer` and `clusterissuer`. `istio-csr`
+is an agent that simplifies the process of obtaining the certificates from the `cert-manager` for
+`OpenShift Service Mesh`. `cert-manager-operator` manages `cert-manager` and extending the operator to manage
+`istio-csr` will help the users to use all the solutions mentioned above effectively and easily.
+
+### User Stories
+
+- As an OpenShift user, I want to have an option to dynamically enable `istio-csr`, so that it can be used only
+  when required.
+- As an OpenShift user, I want to have an option to dynamically configure `istio-csr`, so that only the required
+  features can be enabled.
+
+### Goals
+
+- `cert-manager-operator` to be extended to manage `istio-csr` along with currently managed `cert-manager`.
+- New custom resource `istiocsr` to be made available to enable and configure deployment.
+
+### Non-Goals
+
+- `istio-csr` must be enabled and configured before installing `OpenShift Service Mesh`, which is a
+  limitation of `OpenShift Service Mesh`.
+- Downgrading OpenShift cluster from `istio-csr` supported version to unsupported version is not supported.
+
+## Proposal
+
+`istio-csr` will be managed by `cert-manager-operator` which can be enabled and configured by a user as and
+when required, using new custom resource which will be made available for configuring `istio-csr`.
+
+A new controller will be added to `cert-manager-operator` to manage and maintain the `istio-csr` deployment in
+desired state which can be monitored by user through the status sub-resource of the new custom resource.
+
+`istio-csr` controller will make use of static manifest templates for creating below resources to
+successfully deploy `istio-csr` agent.
+* `certificate.cert-manager.io`
+* `clusterissuer.cert-manager.io`
+* `issuer.cert-manager.io`
+* `clusterrole.rbac.authorization.k8s.io`
+* `clusterrolebinding.rbac.authorization.k8s.io`
+* `deployment`
+* `service`
+* `servicemonitor.monitoring.coreos.com`
+* `namespace`
+* `role.rbac.authorization.k8s.io`
+* `rolebinding.rbac.authorization.k8s.io`
+* `serviceaccount`
+
+`istio-csr` CR object will be a singleton object with the name `default` and an OpenShift user will have to
+create the same for enabling the deployment. `istio-csr` should be deployed in the same namespace as that of
+the `OpenShift Service Mesh` control plane.
+
+When `istiocsr` custom resource is created but `issuerref` is not provided operator will create a
```

Comment:
Can you take a relook at this paragraph? cert-manager Certficiate either uses a cluster scoped issuer (ClusterIssuer) or namespace scoped issuer (referred as only Issuer). There is missing context about the Issuer being created using ClusterIssuer.

Reply:
Yeah it was confusing. Rephrased the content.